wren-ai-service/src/pipelines/generation/question_recommendation.py

```diff
@@ -0,0 +1,250 @@
+import json
+import logging
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import orjson
+from hamilton import base
+from hamilton.experimental.h_async import AsyncDriver
+from haystack.components.builders.prompt_builder import PromptBuilder
+from langfuse.decorators import observe
+from pydantic import BaseModel
+
+from src.core.pipeline import BasicPipeline, async_validate
+from src.core.provider import LLMProvider
+
+logger = logging.getLogger("wren-ai-service")
+
+
+## Start of Pipeline
+@observe(capture_input=False)
+def prompt(
+    mdl: dict,
+    previous_questions: list[str],
+    language: str,
+    num_questions: int,
+    num_categories: int,
+    prompt_builder: PromptBuilder,
+) -> dict:
+    return prompt_builder.run(
+        models=mdl["models"],
```

Comment:
there's still some risk that the mdl is too large. Also I suggest there is room for improvement here on how we structure mdl inside prompt

Reply:
noted: we will trace the token usage in the following future, and consider to thin the models